### PR TITLE
pandasschema

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Added pandasschema for column parsing and slightly prettier/clearer/user-friendliness log for bad public samples (in the sense they all get logged). 

@siebrenf I think the same check was done twice or do I not understand?

**Checklist**
- [x] I made a PR to develop (not master)
- [x] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

File: seq2science/rules/configuration_generic.smk [hunk 1]
```diff
@@ -107,27 +113,35 @@ for key, value in config.items():
 
 # read the samples.tsv file as all text, drop comment lines
 samples = pd.read_csv(config["samples"], sep='\t', dtype='str', comment='#')
-samples.columns = samples.columns.str.strip()
```
Comment (siebrenf) [inline seq2science/rules/configuration_generic.smk:110]: Is it really necessary to crash the pipeline for an extra white space?

File: seq2science/rules/configuration_generic.smk [hunk 2]
```diff
@@ -107,27 +113,35 @@ for key, value in config.items():
...
+allowed_pattern = r'[A-Za-z0-9_.\-%]+'
+column_schema["sample"] = [LeadingWhitespaceValidation(),
+                           TrailingWhitespaceValidation(),
+                           IsDistinctValidation(),
+                           MatchesPatternValidation(allowed_pattern)]
+
+column_schema["descriptive_name"] = [LeadingWhitespaceValidation(),
+                                     TrailingWhitespaceValidation(),
+                                     IsDistinctValidation(),
+                                     MatchesPatternValidation(allowed_pattern)]
```
Comment (siebrenf) [inline seq2science/rules/configuration_generic.smk:131]: all columns are checked in the for loop after this, so maybe make a for loop for these two columns to check the `IsDistinceValidation()`?

Comment (Maarten-vd-Sande) [reply seq2science/rules/configuration_generic.smk:131]: Yep makes sense!

File: seq2science/rules/configuration_generic.smk (hunk 1 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/configuration_generic.smk:110]: Yeah maybe you are right, it's a matter of preference I guess.

**Comment by Maarten-vd-Sande** [comment]
Waiting for this to be accepted: https://github.com/TMiguelT/PandasSchema/pull/37

File: seq2science/rules/configuration_generic.smk [hunk 3]
```diff
@@ -102,33 +104,36 @@ for key, value in config.items():
...
-    "percentage signs (%), underscores (_), periods (.), or minuses (-).\n")
+# use pandasschema for checking if samples file is filed out correctly
+allowed_pattern = r'[A-Za-z0-9_.\-%]+'
+distinct_columns = ["sample"]
+if "descriptive_name" in samples.columns:
+    distinct_columns.append("descriptive_name")
+
+distinct_schema = Schema(
+    [Column(col, [MatchesPatternValidation(allowed_pattern),
+                  IsDistinctValidation(ignore_nan=True)] if col in distinct_columns else [MatchesPatternValidation(allowed_pattern)], allow_empty=True) for col in
```
Comment (siebrenf) [inline seq2science/rules/configuration_generic.smk:126]: `ignore_nan` doesnt exist now right?

Comment (Maarten-vd-Sande) [reply seq2science/rules/configuration_generic.smk:126]: Good catch! The cat :cat: didnt let me sleep last night

**Pull request merged**